app/services/test.py

```diff
@@ -154,18 +272,44 @@ def get_care_plan_names(self, doctor: Doctor) -> InfoCarePlan:
         log(log.INFO, "get_care_plan_names: Count of names [%d]", len(care_plan_names))
         return care_plan_names
 
+    def get_frequency_names(self, doctor: Doctor) -> InfoFrequency:
+        frequency_plan_names = InfoFrequency.query.all()
+        if not frequency_plan_names:
+            log(log.INFO, "get_frequency_names: No care plan names")
+            return
+        log(
+            log.INFO,
+            "get_frequency_names: Count of frequency names [%d]",
+            len(frequency_plan_names),
+        )
+        return frequency_plan_names
+
     def get_test(self, test_id: str, doctor: Doctor) -> GetTest:
-        test: Test = Test.query.filter(Test.id == test_id).first()
+        id = int(test_id)
+        if not id:
+            log(log.INFO, "get_test: No test id")
+            return
+        test: Test = Test.query.filter(Test.id == id).first()
         if not test:
             log(log.INFO, "get_test: No test")
             return
         log(log.INFO, "get_test: Test [%d]", test.id)
 
+        care_plan: CarePlan = CarePlan.query.filter(
+            CarePlan.id == test.care_plan_id
+        ).first()
```

Comment:
can we just use here CarePlan.query.get(test.care_plan_id)?

Reply:
theoretically possible, but here to be exactly as they say)